python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -76,14 +76,9 @@ def __init__(
         self.stream_lock = asyncio.Lock()
 
         self.client: tcpip.Client | None = None
-        self.read_loop_task = utils.make_done_future()
-
         self.mock_data_server: MockSiglentSSA3000xDataServer | None = None
-
-        self.read_data = False
-
-        # Number of consecutive read timeouts encountered.
-        self.num_consecutive_read_timeouts = 0
+        self.do_read_data = False
```

Comment:
I renamed the variable to `_have_seen_data` since it only is used internally to this class. I also made sure to reset it to False in the `setup_reading` method (which is run once each time before the read loop is started) and improved the warning message since it referred to variables that no longer exist.